src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -177,7 +177,7 @@ void fill_tensor_from_bytes(ov::Tensor& tensor, py::array& array) {
     auto data = tensor.data<std::string>();
     for (size_t i = 0; i < tensor.get_size(); ++i) {
         const char* ptr = reinterpret_cast<const char*>(buf.ptr) + (i * buf.itemsize);
-        data[i] = std::string(ptr, buf.ndim == 0 ? buf.itemsize : buf.strides[0]);
+        data[i] = std::string(ptr, strnlen(ptr, buf.itemsize));
```

Comment:
Can tests be added e.g. using data (or more cases) from PR description?

What about case when input buffer has special characters inside the string e.g. `Tes\u0000t`? The `strnlen` can stop on first null character and string will be truncated to `Tes`

Reply:
The truncation happens, I put a `do_not_review` label and I'm looking for another solution.